spec/helpers/generic_helper.rb

```
@@ -1,8 +1,12 @@
 # frozen_string_literal: true
 
 module GenericHelper
-  def wait_for(timeout:, &blk)
-    till = Time.now + (timeout.to_f / 1000)
-    sleep 0.001 while Time.now < till && !blk.call
+  def wait_for(timeout:, sleep_duration: 0.001, message: nil, &blk)
+    till = (Time.now + timeout).to_i
+
+    while Time.now.to_i < till && !blk.call
+      sleep sleep_duration
+      raise message || "Timeout after #{timeout} second" if Time.now.to_i >= till
```

Also raise an error on timeout instead of silently passing.